DatadogFlags/Sources/Client/FlagsClient.swift

```diff
@@ -211,18 +211,71 @@ extension FlagsClient: FlagsClientProtocol {
             reason: flagAssignment.reason
         )
 
-        if let context = repository.context {
-            exposureLogger.logExposure(
-                for: key,
-                assignment: flagAssignment,
-                evaluationContext: context
-            )
-            rumFlagEvaluationReporter.sendFlagEvaluation(
-                flagKey: key,
-                value: value
+        trackEvaluation(key: key)
+
+        return details
+    }
+
+    @_spi(Internal)
+    public func getFlagsDetails() -> [String: FlagDetails<AnyValue>]? {
+        guard let flagAssignments = repository.flagAssignments() else {
+            return nil
+        }
+
+        var result: [String: FlagDetails<AnyValue>] = [:]
+
+        for (key, assignment) in flagAssignments {
+            let value: AnyValue
+
+            switch assignment.variation {
+            case .boolean(let v):   value = .bool(v)
+            case .string(let v):    value = .string(v)
+            case .integer(let v):   value = .int(v)
+            case .double(let v):    value = .double(v)
+            case .object(let v):    value = v
+            case .unknown:          continue
+            }
+
+            result[key] = FlagDetails(
+                key: key,
+                value: value,
+                variant: assignment.variationKey,
+                reason: assignment.reason
             )
         }
 
-        return details
+        return result
+    }
+
+    @_spi(Internal)
+    public func trackEvaluation(key: String) {
```

Comment:
Tracking needs to be done with information from the initial evaluation of the flag assignment. Changing the context and, in the future, polling for new config may change the assigned value and/or the context associated with it.
If the context does not change, then this will work correctly, however, it is susceptible to incorrectly pairing context with an assignment.